test/helpers/replicate.js

```diff
@@ -28,4 +28,21 @@ module.exports = async function replicate(person1, person2) {
       }
     }, 100)
   })
+  if (opts.waitUntilMembersOf) {
+    await waitUntilMember(person1, opts.waitUntilMembersOf)
+    await waitUntilMember(person2, opts.waitUntilMembersOf)
+  }
+}
+
+async function waitUntilMember(person, groupId) {
+  let isMember = false
+  while (!isMember) {
```

Comment:
Can you replace this while loop with a for loop that does at most 20 iterations?

It's because if our implementation is broken (like, bob is never actually added to the group), then this is going to do an infinite loop and we'll never get a test failure.

Reply:
i think CI kills it after 10 min? but sure

Reply:
done